enterprise/backend/src/metabase_enterprise/metabot_v3/tools/api.clj

```diff
@@ -589,6 +651,78 @@
                          [:models  [:sequential ::full-table]]]]]
    [:map [:output :string]]])
 
+(mr/def ::search-data-sources-arguments
+  [:and
+   [:map
+    [:keywords {:optional true} [:maybe [:sequential :string]]]
+    [:description {:optional true} [:maybe :string]]
+    [:database_id {:optional true} [:maybe :int]]
+    [:entity_types {:optional true} [:maybe [:sequential [:enum "table" "model" "question" "dashboard" "metric"]]]]
+    [:limit {:optional true, :default 50} [:and :int [:fn #(<= 1 % 100)]]]]
+   [:map {:encode/tool-api-request
+          #(set/rename-keys % {:database_id :database-id})}]])
```

Comment:
why does `:database_id` get kebabed but not `:entity_types`?

Reply:
Good catch. No specific reason here actually. The Clojure convention is to use kebab-case, right? So we'd transform snake that has been used in the API to kebab always or is my understanding wrong?

Reply:
Yeah . I was browsing the code a bit and it seems like there is a bunch of snake case flying around as well. We should probably refactor this